azurerm/provider.go

```diff
@@ -138,6 +138,8 @@ func Provider() terraform.ResourceProvider {
 			"azurerm_automation_schedule":                     resourceArmAutomationSchedule(),
 			"azurerm_autoscale_setting":                       resourceArmAutoScaleSetting(),
 			"azurerm_availability_set":                        resourceArmAvailabilitySet(),
+			"azurerm_azure_firewall":                          resourceArmAzureFirewall(),
+			"azurerm_azure_firewall_network_rule_collection":  resourceArmAzureFirewallNetworkRuleCollection(),
```

Comment:
Although it looks a little unwieldy Azure Firewall is the name of the product, so the naming is consistent. Lots of things have firewalls in Azure so I think this makes it clear exactly which product is being managed.

Reply:
that's a fair point - but in general we ditch the word `Azure` from the front (e.g. Azure Kubernetes Service is `azurerm_kubernetes_cluster` / Azure Managed Database for MySQL becomes `azurerm_mysql_server`); so I think we should ditch it to be consistent with the other resources (the sole exception to this is `azurerm_azuread`, because that's how it's commonly referred too).

Reply:
@tombuildsstuff OK, i've renamed everything to azurerm_firewall